src/idiomatic/leveraging-the-type-system/raii/scope_guards.md

```diff
@@ -0,0 +1,134 @@
+# Scope Guards
+
+A scope guard makes use of the `Drop` trait to run a given closure when it goes
+out of scope.
+
+```rust,editable,compile_fail
+use scopeguard::{ScopeGuard, guard};
+use std::{
+    fs::{self, File},
+    io::Write,
+};
+
+fn conditional_success() -> bool {
+    true
+}
+
+fn main() {
+    let path = "temp.txt";
+    let mut file = File::create(path).expect("cannot create file");
+
+    // Write something to the file
```

Comment:
Consider making the example a bit more realistic: set it up as if we are downloading a file through HTTP, and if the connection gets interrupted, we don't want to leave behind an incomplete file. I'm not asking to add/change code, only to update comments, and names of functions and variables.